agent/agent.go

```diff
@@ -1252,7 +1234,7 @@ func (a *Agent) reloadWatches(cfg *config.RuntimeConfig) error {
 				httpConfig := wp.Exempt["http_handler_config"].(*watch.HttpHandlerConfig)
 				wp.Handler = makeHTTPWatchHandler(a.logger, httpConfig)
 			}
-			wp.LogOutput = a.LogOutput
+			wp.Logger = a.logger.Named("watch")
```

Comment:
There is a file that has all the logger names defined, I think this should go in there as well.

Reply:
I saw that file, but I'm really not sure about that pattern. Why put all the logger name constants in a single file? A centralized constants file like this ends up introducing some unnecessary package dependencies as everything depends on this one package to get a logger name.

Does this come from a long time ago when Consul was much smaller and this type of centralization might have made sense?

If it's to prevent conflicting names, as long as components are well named a conflict in the logger name seems unlikely.

Given the size and large number of features in Consul today, I would expect constants like this to be in the package where the logger is used. But if a string is only used once, do we need a constant for it?

Reply:
I agree with you, that if a name is only used once, there is not much benefit of putting it into a file. If I remember correctly this file was introduced while moving to hclog -- and that was not that long ago.